File: src/test/resources/integration_tests/python/red_fail_integration_test.py

import java.lang.Exception
import java.lang.RuntimeException
import java.lang.Throwable
import java.lang.Error
from org.vertx.testtools import VertxAssert
import vertx_tests
from core.event_bus import EventBus
import vertx

def test_throw_python_exception():
    raise Exception('foo')

def test_throw_java_exception():
    raise java.lang.Exception('foo')

def test_throw_java_runtime_exception():
    raise java.lang.RuntimeException('foo')

def test_throw_java_throwable():
    raise java.lang.Throwable('foo')

def test_throw_java_error():
    raise java.lang.Error('foo')

def test_test_error():
    # Deliberate language error in test
    print x




vertx_tests.start_tests(locals())

File: src/test/resources/integration_tests/javascript/redFailIntegrationTest.js

var container = require("vertx/container")
var vertx = require("vertx");
var vertxTests = require("vertx_tests");
var vassert = require("vertx_assert");

function testThrowRuntimeDirect() {
  throw new java.lang.RuntimeException("throw runtime");
}

function testThrowCheckedDirect() {
  throw new java.lang.Exception("throw runtime");
}

function testThrowErrorDirect() {
  throw new java.lang.Error("throw runtime");
}

function testThrowThrowableDirect() {
  throw new java.lang.Throwable("throw runtime");
}

function testThrowString() {
  throw "wibble"
}

function testTestError() {
  // Should give error - x is not defined
  if (x == 3) {
    console.log("foo")
  }
}

vertxTests.startTests(this);




